src/load/azext_load/data_plane/load_trigger/custom.py

```diff
@@ -0,0 +1,233 @@
+        description=description,
+    )
+    logger.debug("Trigger schedule body: %s", trigger_body)
+    try:
+        response = client.create_or_update_trigger(trigger_id=trigger_id, body=trigger_body)
+        logger.debug("Created trigger schedule: %s", response)
+        logger.info("Creating trigger schedule completed")
+        return response
+    except Exception as e:
+        logger.error("Error occurred while creating trigger schedule: %s", str(e))
+
+
+def update_trigger_schedule(
+    cmd,
+    load_test_resource,
+    trigger_id,
+    resource_group_name=None,
+    description=None,
+    display_name=None,
+    trigger_start_date_time=None,
+    recurrence_type=None,
+    recurrence_interval=None,
+    recurrence_index=None,
+    recurrence_cron_expression=None,
+    recurrence_dates_in_month=None,
+    recurrence_week_days=None,
+    end_after_occurrence=None,
+    end_after_date_time=None,
+    test_ids=None,
+):
+    client = get_admin_data_plane_client(cmd, load_test_resource, resource_group_name)
+    logger.info("Updating trigger schedule with name")
+    existing_trigger_schedule: models.ScheduleTestsTrigger = None
+    try:
+        existing_trigger_schedule = client.get_trigger(trigger_id)
+    except Exception as e:
+        logger.debug("Error occurred while fetching trigger schedule: %s", str(e))
+        logger.error("Trigger schedule with id: %s not found.", trigger_id)
+        return
+    logger.debug("Existing trigger schedule: %s", existing_trigger_schedule)
+    recurrence_end_body = utils.get_recurrence_end_body(
+        end_after_occurrence,
+        end_after_date_time,
+        existing_trigger_schedule.recurrence.recurrence_end if existing_trigger_schedule.recurrence else None
+    )
+    logger.debug("Recurrence end object: %s", recurrence_end_body)
+    recurrence_body = utils.get_recurrence_body_for_update(
+        recurrence_type,
+        recurrence_interval,
+        recurrence_index,
+        recurrence_cron_expression,
+        recurrence_dates_in_month,
+        recurrence_week_days,
+        recurrence_end_body,
+        existing_trigger_schedule.recurrence
+    )
+    logger.debug("Recurrence object: %s", recurrence_body)
+    new_trigger_body = utils.get_schedule_trigger_body_for_update(
+        existing_trigger_schedule,
+        recurrence_body,
+        display_name,
+        description,
+        trigger_start_date_time,
+        test_ids,
+    )
+    
+    logger.debug("Trigger schedule body: %s", new_trigger_body)
+    logger.debug("Trigger schedule body to be sent for update: %s", existing_trigger_schedule)
+    try:
+        response = client.create_or_update_trigger(trigger_id=trigger_id, body=new_trigger_body)
+        logger.debug("Updated trigger schedule: %s", response)
+        logger.info("Updating trigger schedule completed")
+        return response
+    except Exception as e:
+        logger.error("Error occurred while updating trigger schedule: %s", str(e))
```

Comment:
Why are we not throwing errors here?

Reply:
Error occurred while updating schedule trigger